torch/_inductor/codegen/cuda/gemm_template.py

```diff
@@ -1374,9 +1374,9 @@ def _are_inputs_layout_compatible(self, layouts: List[Layout]) -> bool:
             return False
         A_size = [int(i) for i in A_layout.size]
         B_size = [int(i) for i in B_layout.size]
-        K = max(A_size[-1], B_size[-2])
         M = A_size[-2]
         N = B_size[-1]
+        K = max(M, N)
```

Comment:
Hold on. Indices swapped, these two statements are not equivalent!

Reply:
Oof, what an ineptitude on my part. I simply got rid of the unused variables. The logic around these lines of code is unnecessarily complicated but I shouldn't get outside my lane...